notebooks/learning-to-rank/01-learning-to-rank.ipynb

```diff
@@ -0,0 +1,1556 @@
+        "judgments_df"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {},
+      "source": [
+        "## Configure feature extraction\n",
+        "\n",
+        "Features are the input data of our model. They represent the document in the context of the query.\n",
+        "Features are configured using templated queries to extract features.\n",
+        "\n",
+        "To define features extraction, you will be using the primitives provided by the Eland API:"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": 9,
+      "metadata": {
+        "id": "LjxAj4lQqEYJ"
+      },
+      "outputs": [],
+      "source": [
+        "from eland.ml.ltr import LTRModelConfig, QueryFeatureExtractor\n",
+        "\n",
+        "ltr_config = LTRModelConfig(\n",
+        "  feature_extractors = [\n",
+        "      # For the following field we want to use the score of the match query for the field as a features:\n",
+        "      QueryFeatureExtractor(\n",
+        "        feature_name=\"title_bm25\",\n",
+        "        query={ \"match\": { \"title\": \"{{query}}\" } }\n",
+        "      ),\n",
+        "      QueryFeatureExtractor(\n",
+        "        feature_name=\"actors_bm25\",\n",
+        "        query={ \"match\": { \"actors\": \"{{query}}\" } }\n",
+        "      ),\n",
+        "      # We could also use a more strict matching clause as an additional features. Here we want all the terms of our query to match.\n",
+        "      QueryFeatureExtractor(\n",
+        "        feature_name=\"title_all_terms_bm25\",\n",
+        "        query={ \"match\": { \"title\": { \"query\": \"{{query}}\", \"minimum_should_match\": \"100%\" } } }\n",
+        "      ),\n",
+        "      # Also we can use a script_score query to get the document field values directly as a feature.\n",
+        "      QueryFeatureExtractor(\n",
+        "        feature_name=\"popularity\",\n",
+        "        query={\n",
+        "            \"script_score\": {\n",
+        "              \"query\": { \"exists\": { \"field\": \"popularity\" } },\n",
+        "              \"script\": { \"source\": \"return doc['popularity'].value;\" }\n",
+        "          }\n",
+        "        }\n",
+        "      )\n",
+        "  ]\n",
+        ")"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {},
+      "source": [
+        "## Adding features to the judgement list\n",
+        "\n",
+        "During this step we will add features to our judgmennt list. The resuling dataframe will be used to train our model.\n",
+        "\n",
+        "**Note** This operation is quite fast if your Elasticsearch instance is local or close (around 1 min 30 sec.) but can be much longer if it is not the case (more than 10 minutes). When using Google Collab it is difficult to control where your notebook is executed and it is likely that you can get in the later case."
```

Comment:
Suggested change:

> ## Building the training dataset
> Now that we have our basic datasets loaded, and feature extraction configured, we'll use our judgement list to come up with the final dataset for training. The dataset will consist of rows containing `<query, document>` pairs, as well as all of the features we need to train the model. To generate this dataset, we'll run each query from the judgement list and add the extracted features as columns for each of the labelled result documents in the judgement list.
> For example, if we have a query `q1` with two labelled documents `d3` and `d9`, the training dataset will end up with two rows — one for each of the pairs `<q1, d3>` and `<q1, d9>`.
> Note that because this executes queries on your Elasticsearch cluster, the time to run this operation will vary depending on where the cluster is versus where this notebook runs. For example, if you run the notebook on the same server or host as the Elasticsearch cluster, this operation tends to run very quickly on the sample dataset (< 2 mins).

Reply:
Applied your suggestion.